inception/inception-api-annotation-experimental/src/main/java/de/tudarmstadt/ukp/inception/experimental/api/AnnotationSystemAPIImpl.java

```diff
@@ -0,0 +1,418 @@
+
+    }
+
+    @Override
+    public void handleSelectAnnotation(ClientMessage aClientMessage) throws IOException
+    {
+        CAS cas = getCasForDocument(aClientMessage.getUsername(), aClientMessage.getProject(),
+                aClientMessage.getDocument());
+        AnnotationFS annotation = selectAnnotationByAddr(cas,
+                aClientMessage.getAnnotationAddress());
+        AnnotationMessage message = new AnnotationMessage(String.valueOf(annotation._id()),
+                annotation.getBegin(), annotation.getEnd(), annotation.getType().getShortName(),
+                annotation.getCoveredText());
+        annotationProcessAPI.handleSendSelectAnnotation(message, aClientMessage.getClientName());
+    }
+
+    @Override
+    public void handleCreateAnnotation(ClientMessage aClientMessage) throws IOException
+    {
+        try {
+            CasStorageSession.open();
+
+            MDC.put(Logging.KEY_REPOSITORY_PATH, repositoryProperties.getPath().toString());
+            SourceDocument sourceDocument = documentService
+                    .getSourceDocument(aClientMessage.getProject(), aClientMessage.getDocument());
+
+            CAS cas = documentService.readAnnotationCas(sourceDocument,
+                    aClientMessage.getUsername());
+
+            Type type = null;
+            for (Iterator<Feature> it = cas.getTypeSystem().getFeatures(); it.hasNext();) {
+
+                Feature s = it.next();
+                if (s.getDomain().getShortName().equals(aClientMessage.getAnnotationType())) {
+                    type = s.getDomain();
+
+                }
+            }
+
+
+            TypeAdapter adapter = annotationService
+                    .getAdapter(annotationSystemAPIService.getAnnotationLayer(type.getName()));
+            AnnotationFS newAnnotation = null;
+            if (adapter instanceof SpanAdapter) {
+                newAnnotation = ((SpanAdapter) adapter).add(
+                        documentService.getSourceDocument(aClientMessage.getProject(),
+                                aClientMessage.getDocument()),
+                        aClientMessage.getUsername(), cas,
+                        aClientMessage.getAnnotationOffsetBegin(),
+                        aClientMessage.getAnnotationOffsetEnd());
+            }
+
+            //TODO more adapater instances
+
+            CasStorageSession.get().close();
+
+            AnnotationMessage message = new AnnotationMessage();
+            message.setAnnotationAddress(String.valueOf(newAnnotation.getAddress()));
+            message.setAnnotationOffsetBegin(newAnnotation.getBegin());
+            message.setAnnotationOffsetEnd(newAnnotation.getEnd());
+            message.setAnnotationText(newAnnotation.getCoveredText());
+            message.setAnnotationType(newAnnotation.getType().getShortName());
+            annotationProcessAPI.handleSendUpdateAnnotation(message,
```

Comment:
Why do you manually generate a response here instead of translating the `SpanCreatedEvent` that is created by the SpanAdapter and broadcast in the application to a socket message?

We have a number of events already related to annotation creation/updating/deletion - and since a single call to an adapter method can generate any number of these, you should have to observe these messages and forward them to the client instead of believing that the request you got from the client has only a single effect and that you know what that effect is.